---
title: "PairwiseCI_X-A"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Pairwise CI
This code will read the RNA-seq TPM file and calculate the X:A ratio with confident interval by pairwiseCI 

```{r load library}
# install these library first if it is not in your library
library(plyr)
library(lattice)
library(reshape)
#install.packages("MCPAN")
library(MCPAN)
#install.packages("pairwiseCI")
library(pairwiseCI)

```

```{r load data }
p=list.files("/Users/ellie/Downloads/RNA-seq",full.names = TRUE, pattern = "avg.txt") #it is just how I have files name ends by avg.txt
```

```{r read data and bootstrap CI}


for (i in 1:length(p)){
  #read data
  data = read.table(p[i], header = FALSE)
  names(data) = c("sample_name","gene","chr","value") # give sample header line
  
  data$Xchr = factor(ifelse(data$chr=="AC_000187.1","X","A")) 
  #if chr = AC_000187.1, it is X, others is A. change to "chrX" if this is the name in genome
 
   
  #Bootstrap CI
  CI=pairwiseCI(value~Xchr,by="sample_name",data = data,alternative = "two.sided", conf.level = 0.95, method = "Median.ratio")
  
  
  png(paste(unlist(strsplit(p[i],".txt"))[1],".png",sep = "")) # this plot is optional since CI value will plot in ggplot
  plot(CI)
  dev.off()

  sink(paste(unlist(strsplit(p[i],".txt"))[1],"_CI_output.txt",sep = "")) # this output is needed for ggplot
  print(CI)
  sink()
  
  
}
```



```{r ggplot2}
library(ggplot2)
# this data is the same from "_CI_output.txt"
d=data.frame(Stage=c("MII","2C","4C","8C","16C","32C","CM","BL",
                     "D7","D10","D13","D16","D19"), 
             median=c(0.8961, 0.8801,0.8325,1.103,1.63, 1.509, 1.033, 1,
                      1.044,1.181,1.194,1.172,1.14),
             lower=c(0.6701,0.5646,0.5009,0.9086,1.2,0.9337,0.6839,0.688,
                     0.8385,0.9153,0.866,0.7974,0.8638), 
             upper=c(1.735,1.397,1.361,1.793,2.013,2.061,1.337,1.317,
                     1.445,1.613,1.659,1.727,1.532 ))

d$order=factor(d$Stage, levels = c("MII","2C","4C","8C","16C","32C","CM","BL",
                                   "D7","D10","D13","D16","D19"),
               ordered = TRUE)

tiff("Vivo embryos ubiquitous genes.tiff", width = 8, height = 5, units = 'in', res = 300)
ggplot(d, aes(x=order,y=median,group = 1)) + 
  scale_y_discrete(limit=c(0.0,0.5,1,1.5,2.0))+
  geom_line() + geom_hline(yintercept=1, color="light blue")+
  geom_hline(yintercept=0.5, color="red")+
  geom_point() +
  geom_pointrange(data=d, mapping=aes(y=median, ymin=upper, ymax=lower),color='blue', size=1) + 
  labs(x = "Embryo developmental stages", y = "X:A ratio")+
  theme(panel.background = element_blank(),
        axis.line = element_line(colour = "black"),
        axis.text=element_text(size=12),
        axis.title=element_text(size = 14,face = "bold")) #remove the background
dev.off()

#if want the background with lines
#panel.grid.major.x  = element_line(colour = "light grey")
#panel.background = element_rect(fill="white")
```